woocommerce/woocommerce
# CYS - Core - Check the Font Library availability

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44076 .



This PR adds the `isFontLibraryAvailable` variable in the context of the main state machine. This guarantees that all the children's machines can access this variable to implement logic based on the availability of the Font Library.

This PR includes a hacky approach to avoid duplicate GET requests: https://github.com/woocommerce/woocommerce/blob/c5b0089d084b41a51be2f9e36795c3dfd4c8d966/plugins/woocommerce-admin/client/customize-store/index.tsx/#L541-L543.

The reason is that the `customize-store` bundle is loaded when the client requests URLs such as /customize-store/* ([source code](https://github.com/woocommerce/woocommerce/blob/7867c8bc742106482e3c2e43de27d399448ea389/plugins/woocommerce-admin/client/layout/controller.js/#L330-L347)). However, the assembler-hub is loaded into an iframe with the URL `/customize-store/assembler-hub` ([source code](https://github.com/woocommerce/woocommerce/blob/e7884409761bc8e404cd80208b078f00e0c886f0/plugins/woocommerce-admin/client/customize-store/design-without-ai/actions.ts/#L18-L19)). This means that the entire Customize Store app is loaded again into the iframe. For this reason, the state machines in the iframe start with the initial state. The function `setFlagsForIframeInstance` is a hacky approach to pass the flags from the main instance to the iframe instance. 

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
3. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
4. Go to the intro page and ensure that the `wp-json/wp/v2/font-collections` is fetched.
5. Follow the process. Ensure that the `wp-json/wp/v2/font-collections` is fetched once.
6. Click Done.
7. Now, lands on the assembler hub page using the `wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fassembler-hub`.
8. Ensure that the `wp-json/wp/v2/font-collections` is fetched once.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
CYS - Core - Check the Font Library availability.


#### Comment 

</details>
